messybrainz/webserver/views/api.py

```diff
@@ -55,3 +63,111 @@ def get_aka(messybrainz_id):
     (as specified in the clusters table).
     """
     raise NotImplementedError
+
+
+@api_bp.route("/msid", methods=['GET'])
+@crossdomain()
+def get_msid_using_msid():
+    """Returns all MessyBrainz IDs for the MSID in request."""
+
+    msid = request.args.get('id')
+    request_type = request.args.get('request_type')
+    mbid = False
+    includes = request.args.getlist('includes')
+    if 'mbid' in includes:
+        mbid = True
+
+    data = {}
+    try:
+        if not validate_params(msid, request_type, includes):
+            raise BadDataException
+        if request_type == 'artist':
+            with db.engine.begin() as connection:
+                cluster_id = db_artist.get_cluster_id_using_msid(connection, msid)
+                fetched_msids = db_artist.get_msids_using_cluster_id(connection, cluster_id)
+                if mbid:
+                    fetched_mbids = db_artist.get_artist_mbids_using_msid(connection, cluster_id)
+
+        elif request_type == 'recording':
+            with db.engine.begin() as connection:
+                cluster_id = db_recording.get_recording_cluster_id_using_msid(connection, msid)
+                fetched_msids = db_recording.get_gids_using_cluster_id(connection, cluster_id)
+                if mbid:
+                    fetched_mbids = db_recording.get_mbids_using_cluster_id(connection, cluster_id)
+        elif request_type == 'release':
+            with db.engine.begin() as connection:
+                cluster_id = db_release.get_cluster_id_using_msid(connection, msid)
+                fetched_msids = db_release.get_msids_using_cluster_id(connection, cluster_id)
+                if mbid:
+                    fetched_mbids = db_release.get_release_mbids_using_msid(connection, cluster_id)
+
+        data = {'msid': fetched_msids}
+        if mbid:
+            data['mbid'] = fetched_mbids
+    except BadDataException:
+        raise BadRequest
+    except NoDataFoundException:
+        raise NotFound
+
```

Comment:
There are likely other exceptions the DB is going to throw -- we should catch more exceptions here.